### PR TITLE
Feature/using ts supertype

### PR DESCRIPTION
**Pull request by sshankar19**
Semotus using the new version of supertype

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -505,6 +506,7 @@ RemoteObjectTemplate.processMessage = function processMessage(remoteCall, subscr
         }
     }
 
+    // debugger;
```
Comment (hackerrdave) [inline index.js]: remove commented debugger (multiple places)

File: index.js [hunk 2]
```diff
@@ -34,9 +34,10 @@
  else {
         root.RemoteObjectTemplate = factory(root.Q, root._, root.ObjectTemplate);
     }
-}(this, function (Q, _, ObjectTemplate) {
+}(this, function (Q, _, ObjectTemplate2) {
     'use strict';
-
+    
+var ObjectTemplate = ObjectTemplate2.default;
```
Comment (hackerrdave) [inline index.js]: pending discussion regarding default export

File: index.js [hunk 3]
```diff
@@ -1146,6 +1149,7 @@ RemoteObjectTemplate._setupProperty = function setupProperty(propertyName, defin
 
            return function z() {
 
+                //** Don't think I need to change this */
```
Comment (hackerrdave) [inline index.js]: remove this comment? (all locations)

File: index.js [hunk 4]
```diff
@@ -2629,20 +2639,21 @@ RemoteObjectTemplate.bindDecorators = function (objectTemplate) {
 
 	    // Called first time with parameter rather than target - call supertypes supertypeClass function which will
         // return a function that must be called on the 2nd pass when we have a target.  It will remember parameter
-        ret = ObjectTemplate.supertypeClass(target, objectTemplate);
+        ret = ObjectTemplate2.supertypeClass(target, objectTemplate);
```
Comment (hackerrdave) [inline index.js]: should this be `ObjectTemplate` or `ObjectTemplate2` ?

Comment (sshankar19) [reply index.js]: Hmm, I should rename this. `ObjectTemplate2` is the entire "SupertypeModule", the whole shebang. Part of the work for typescripting Supertype is splitting it out into different components, so now we have an `ObjectTemplate` class, which is basically the `amorphicStatic` reference. 

In this snippet of code I'm storing the module of supertype (e.g. `require('supertype')` in a variable called `ObjectTemplate2`. If I want to access the exported `supertypeClass` property from the supertype module, I can access it like `ObjectTemplate2.supertypeClass`. This is similar to just 
```
var _ = require('underscore');
_.find(...)
```
or whatever.

`ObjectTemplate` in this snippet of code refers to the definition of the _root amorphic object itself_. This is an actual object, a real construct.

I will rename `ObjectTemplate2` to `SupertypeModule` to be more explicit about what it is. I meant to do so beforehand but completely forgot. Nice catch.

**Review by srksag-zz — APPROVED** [review]

File: test/semotus.js [hunk 5]
```diff
@@ -4,16 +4,19 @@
  *
  */
 
+// debugger;
```
Comment (hackerrdave) [inline test/semotus.js]: remove commented debugger (both places)

**Review by hackerrdave — CHANGES_REQUESTED** [review]

**Review by hackerrdave — APPROVED** [review]

**Review by nsaenz7 — APPROVED** [review]

**Review by nikmash — APPROVED** [review]
Looks good

**Pull request merged**